datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -137,6 +179,9 @@ fn is_one(s: &Expr) -> bool {
         | Expr::Literal(ScalarValue::UInt64(Some(1))) => true,
         Expr::Literal(ScalarValue::Float32(Some(v))) if *v == 1. => true,
         Expr::Literal(ScalarValue::Float64(Some(v))) if *v == 1. => true,
+        Expr::Literal(ScalarValue::Decimal128(Some(v), _p, _s)) => {
+            *_s < DECIMAL128_MAX_PRECISION && POWS_OF_TEN[*_s as usize] == *v
```

Comment:
Question: Is there a specific reason to embed a `POWS_OF_TEN` array instead of dynamically calculating it at the runtime? E.g.

```rust
i128::pow(10, *_s as u32) == *v
```

Reply:
pow call make the patch lesser but will do a real multiplication instead of a pointer deref.
This is perf degradaton IMHO. In the worst case there will be 6 such multiplications(2 for Multiply, 2 for Divide, 2 for Modulo) and there might be multiple scalars binary ops in the expression.

Reply:
On the other side I don't see much benefit from using pow except there will be no static array. Not a big profit given a potential perf gain.

Reply:
What is your argument in support for using pow() though?

Reply:
I think this code will run ~1 per query / expression, so I don't expect the any performance difference to be noticeable.

For what it is worth, I think both approaches are reasonable. Thank you both for the discussion.

Reply:
If the function or the logic will be call many time in the runtime, it is better to use static value or const value which will get good performance.